routing/index_router.cpp

```diff
@@ -302,62 +314,52 @@ IRouter::ResultCode IndexRouter::ProcessLeaps(vector<Segment> const & input,
 
   WorldGraph & worldGraph = starter.GetGraph();
   WorldGraph::Mode const worldRouteMode = worldGraph.GetMode();
-  worldGraph.SetMode(WorldGraph::Mode::SingleMwm);
+  worldGraph.SetMode(WorldGraph::Mode::NoLeaps);
```

Comment:
This function depends on the fact that worldGraph mode was not modified before the call and the same as mode variable in DoCalculateRoute. I suggest you to get rid of this assumption and directly pass worldGraphMode as a parameter to the function.